crates/database/src/fee_policies.rs

```diff
@@ -72,6 +73,33 @@ pub async fn fetch(
     Ok(rows)
 }
 
+pub async fn fetch_all_for_auction(
+    ex: &mut PgConnection,
+    auction_id: AuctionId,
+) -> Result<HashMap<OrderUid, Vec<FeePolicy>>, sqlx::Error> {
+    const QUERY: &str = r#"
+    SELECT * FROM fee_policies
+    WHERE auction_id = $1
+    ORDER BY order_uid, application_order
+    "#;
+
+    let rows: Vec<FeePolicy> = sqlx::query_as::<_, FeePolicy>(QUERY)
+        .bind(auction_id)
+        .fetch_all(ex)
+        .await?;
+
+    let mut grouped_fee_policies: HashMap<OrderUid, Vec<FeePolicy>> = HashMap::new();
```

Comment:
Shouldn't this logic be in the caller since this is just a database query function? :thinking:

Reply:
I don't think it's meaningful to use this data without grouping in any way, so I put it there.